server/src/main/java/org/elasticsearch/transport/netty4/Netty4InboundStatsHandler.java

```diff
@@ -93,4 +87,15 @@ public void close() {
         openChannels.clear();
     }
 
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        if (msg instanceof ByteBuf bb) {
+            statsTracker.incrementBytesReceived(bb.readableBytes());
+            statsTracker.incrementMessagesReceived();
+        } else {
+            logger.warn("Message sent is: {} and not a ByteBuf, cannot track received bytes or message count",
+                msg.getClass().getCanonicalName());
```

Comment:
Actually, we could always track message count. When can we end up here?

Reply:
I really don't know it was recommended by @mfussenegger, and I agree to keep a warn message , to be aware that the msg is not a `ByteBuf`. I don't think it's bad to have the phrase `cannot track...` it can help to look up such an instance and act, if this logic breaks in the future.